GG/src/ListBox.cpp

```diff
@@ -853,20 +853,42 @@ void ListBox::PreRender()
                 NormalizeRow(row);
     }
 
-    AdjustScrolls(false);
+    // Adding/removing scrolls and prerendering rows may change the row sizes and require a change
+    // in added/removed scrolls. This may not be stable. Perform two cycles and if it is not
+    // stable then force the scrollbar to be added if either cycle had a scroll bar.
 
-    // Reset require prerender after call to adjust scrolls
-    Control::PreRender();
+    // Perform a cycle of adjust scrolls and prerendering rows and return if sizes changed.
+    auto check_adjust_scroll_size_change =
+        [this](std::pair<bool, bool> force_scrolls = {false, false})
+        {
+            // This adjust scrolls may add or remove scrolls
```

Comment:
I am unsure of your meaning.  What I see is,
````
auto variable_name =
    []()  // this line is indented because it is a continued line of code
    {
        AdjustScrolls()  //this  line is indented because it is the first line of code within the braced compond statement '{'
        ...
    };  // end of continue line of code.
````